docs/vmanomaly - release 1.18.5 (#7684)
VictoriaMetrics/VictoriaMetrics: docs/anomaly-detection/FAQ.md

## Scaling vmanomaly

> **Note:** As of latest release we do not support cluster or auto-scaled version yet (though, it's in our roadmap for - better backends, more parallelization, etc.), so proposed workarounds should be addressed *manually*.

`vmanomaly` supports **vertical** scalability, benefiting from additional CPU cores (resulting in faster processing times) and increased RAM (allowing more models to be trained and larger volumes of timeseries data to be processed efficiently).

For **horizontal** scalability, `vmanomaly` can be deployed as multiple independent instances, each configured with its own [MetricsQL](https://docs.victoriametrics.com/metricsql/) queries and [configurations](https://docs.victoriametrics.com/anomaly-detection/components/):

> - By splitting **queries** [defined in the reader section](https://docs.victoriametrics.com/anomaly-detection/components/reader#vm-reader) and assigning each subset to a separate service instance. If you have *a single query returning a large number of timeseries*, you can further split it by applying global MetricsQL filters using the `extra_filters` [parameter in the reader](https://docs.victoriametrics.com/anomaly-detection/components/reader?highlight=extra_filters#vm-reader). See the example below.

Reviewer: ```suggestion
- Split by ***query*** should be used if you have *a single query returning a large number of timeseries*. Split by query is [explained further in the reader section](https://docs.victoriametrics.com/anomaly-detection/components/reader#vm-reader). This can be further split by applying global MetricsQL filters using the `extra_filters` [parameter in the reader](https://docs.victoriametrics.com/anomaly-detection/components/reader?highlight=extra_filters#vm-reader). See example below.

> - By splitting **models** (e.g., if you decide to run multiple models per timeseries for use cases such as averaging anomaly scores in your alerting rules with `vmalert`, or employing a voting approach to reduce false positives). See the `queries` argument in the [model configuration](https://docs.victoriametrics.com/anomaly-detection/components/models#queries). Additionally, this approach is useful when you just have a large set of resource-intensive independent models.

Reviewer: ```suggestion
- Spliting by **model** should be used when running multiple models on the same query. This is commonly done to reduce false positives by alerting only if multiple models detect an anomaly. See the `queries` argument in the [model configuration](https://docs.victoriametrics.com/anomaly-detection/components/models#queries). Additionally, this approach is useful when you just have a large set of resource-intensive independent models.
```

> - By splitting **schedulers** (e.g., if you need the same models to be trained or inferred under different schedules). Refer to the `schedulers` argument in the [model section](https://docs.victoriametrics.com/anomaly-detection/components/models#schedulers) and the `scheduler` [component documentation](https://docs.victoriametrics.com/anomaly-detection/components/scheduler).

Reviewer: ```suggestion
- Splitting by **scheduler** should be used when the same models needs to be trained or inferred under different schedules. Refer to the `schedulers` argument in the [model section](https://docs.victoriametrics.com/anomaly-detection/components/models#schedulers) and the `scheduler` [component documentation](https://docs.victoriametrics.com/anomaly-detection/components/scheduler).
```

### Splitting the config

> Starting from [v1.18.5](https://docs.victoriametrics.com/anomaly-detection/changelog/#v1185), a CLI utility `config_splitter.py` is available. This tool enables splitting a parent YAML configuration file into multiple sub-configurations based on logical entities (`schedulers`, `queries`, `models`, `extra_filters`). The resulting sub-configurations are fully validated and functional, accounting for many-to-many relationships between models, their associated queries, and the schedulers they are linked to. These sub-configurations can then be saved to a specified directory for further use:

Reviewer: ```suggestion
Starting from [v1.18.5](https://docs.victoriametrics.com/anomaly-detection/changelog/#v1185), a CLI utility named `config_splitter.py` is available in vmanoamly. The config splitter tool enables splitting a parent vmanomaly YAML configuration file into multiple sub-configurations based on logical entities  such as `schedulers`, `queries`, `models`, `extra_filters`. The resulting sub-configurations are fully validated, functional, account for many-to-many relationships between models and their associated queries, and the schedulers they are linked to. These sub-configurations can then be saved to a specified directory for further use:
```
